### PR TITLE
Update authorization client error search and handle 500 errors

### PR DESCRIPTION
**Pull request by JuanCrg90**
This updates the authorization client to use the same approach that api client uses to search for the error class. This adds a new ErrorParser class that is in charge to verify which error class should be returned. Also this adds a status code validation to raise a EasyMeli exception in server side errors.

### CODE REVIEW

File: lib/easy_meli/errors.rb [hunk 1]
```diff
@@ -73,4 +73,23 @@ def local_message
...
+    ERROR_LIST = {
+      'Error validating grant' => EasyMeli::InvalidGrantError,
+      'The User ID must match the consultant\'s' => EasyMeli::ForbiddenError,
+      'invalid_token' => EasyMeli::InvalidTokenError,
+      'Malformed access_token' => EasyMeli::MalformedTokenError,
+      'too_many_requests' => EasyMeli::TooManyRequestsError,
+      'unknown_error' => EasyMeli::UnknownError
+    }
+
+  def self.error_class(body)
```
Comment (enortham) [inline lib/easy_meli/errors.rb]: Bad indents but I would also move this into its own file. I think this file should be only for errors.

**Review by enortham — CHANGES_REQUESTED** [review]
I notice that you're sending lots of versions. FYI you can test the gem locally with real data to make sure it works as expected.

File: lib/easy_meli/error_parser.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  ERROR_LIST = {
+    'Error validating grant' => EasyMeli::InvalidGrantError,
+    'The User ID must match the consultant\'s' => EasyMeli::ForbiddenError,
+    'invalid_token' => EasyMeli::InvalidTokenError,
+    'Malformed access_token' => EasyMeli::MalformedTokenError,
+    'too_many_requests' => EasyMeli::TooManyRequestsError,
+    'unknown_error' => EasyMeli::UnknownError
+  }
+
+  STATUS_ERRORS = {
```
Comment (JuanCrg90) [inline lib/easy_meli/error_parser.rb:11]: I found that some ML server side errors return an empty body. I've created specific errors that inherit from `EasyMeli:: ServerError` to help with the error handling.

**Comment by JuanCrg90** [comment]
@enortham changes addressed and I've added some new error classes for server side errors, I found that some  500 responses are sent without body, I think is better if EasyMeli handles these responses and raises an exception.

File: lib/easy_meli/error_parser.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  def self.error_class(response)
+    ERROR_LIST.find { |key, _| self.error_message_from_body(response)&.include?(key) }&.last
+  end
+
+  def self.status_error_class(response)
+    return unless self.server_side_error?(response)
+
+    STATUS_ERRORS[response.code]
```
Comment (enortham) [inline lib/easy_meli/error_parser.rb]: Maybe you should some general sort of error if the code doesn't match anything.

**Comment by JuanCrg90** [comment]
Updated @enortham what do you think?

**Review by enortham — APPROVED** [review]

**Pull request merged**